stan/math/prim/mat/prob/ordered_logistic_log.hpp

```diff
@@ -51,6 +52,52 @@ namespace stan {
       return ordered_logistic_lpmf<T_lambda, T_cut>(y, lambda, c);
     }
 
+    /**
```

Comment:
The first argument (`int` or `std::vector<int>`) could just be templated out in this file.  It will only be able to be instantiated with `int` or `std::vector<int>` because the function to which it delegates has that restriction.  The doc will have to be generalized a bit.

This would be nice to do, but it's optional for this pull request.